pkg/scheduler/internal/cache/node_tree.go

```diff
@@ -49,8 +51,10 @@ func (na *nodeArray) next() (nodeName string, exhausted bool) {
 		return "", false
 	}
 	if na.lastIndex >= len(na.nodes) {
+		na.exhausted = true
```

Comment:
could you put this in a method instead? then you don't need to store it as a boolean